File: ruby/bob/bob.rb
class Bob

  def hey(remark)
    case 
    when remark == remark.upcase && remark != remark.downcase
      'Whoa, chill out!'
    when remark.end_with?('?')
      'Sure.'
    when remark =~ /\A\s*\z/
      "Fine. Be that way!"
    else
      'Whatever.'
    end
  end

end

File: lua/bob/bob.lua
local bob = {}

function bob.hey(remark)
  local response = ''
  if remark == remark:upper() and remark ~= remark:lower() then
    response = 'Whoa, chill out!'
  elseif remark:match("?$") then
    response = 'Sure'
  elseif not remark:match('%s') then
    response = 'Fine, be that way.' 
  else
    response = 'Whatever'
  end
  return response
end

return bob
